Documentation/policy.rst

```diff
@@ -0,0 +1,408 @@
+        // characters disallowed from the conventional "path" part of a URL as defined
+        // by RFC 3986.
+        type PortRuleHTTP struct {
+                // Path is an extended POSIX regex matched against the path of a
+                // request. Currently it can contain characters disallowed from the
+                // conventional "path" part of a URL as defined by RFC 3986. Paths must
+                // begin with a '/'.
+                //
+                // If omitted or empty, all paths are all allowed.
+                //
+                // +optional
+                Path string `json:"path,omitempty" protobuf:"bytes,1,opt,name=path"`
+
+                // Method is an extended POSIX regex matched against the method of a
+                // request, e.g. "GET", "POST", "PUT", "PATCH", "DELETE", ...
+                //
+                // If omitted or empty, all methods are allowed.
+                //
+                // +optional
+                Method string `json:"method,omitempty" protobuf:"bytes,1,opt,name=method"`
+
+                // Host is an extended POSIX regex matched against the host header of a
+                // request, e.g. "foo.com"
+                //
+                // If omitted or empty, the value of the host header is ignored.
+                //
+                // +optional
+                Host string `json:"host,omitempty" protobuf:"bytes,1,opt,name=method"`
+
+                // Headers is a list of HTTP headers which must be present in the
+                // request. If omitted or empty, requests are allowed regardless of
+                // headers present.
+                //
+                // +optional
+                Headers []string `json:"headers,omitempty"`
+        }
+
+
+Example (HTTP)
+--------------
+
+The following example limits all endpoints which carry the labels
+``app=myService`` to only be able to receive packets on port 80 using TCP.
+While communicating on this port, the only API endpoints allowed will be ``GET
+/path1`` and ``PUT /path2`` with the HTTP header ``X-My_header`` set to
+``true``:
+
+        {
+            "endpointSelector": {"matchLabels":{"app":"myService"}},
+            "ingress": [{
+                "toPorts": [{
+                    "ports": [
+                        {"port": "80", "protocol": "tcp"}
+                    ],
+                    "rules": {
+                        "HTTP": [
+                            {
+                                "method": "GET",
+                                "path": "/path1$"
+                            },{
+                                "method": "PUT",
+                                "path": "/path2$",
+                                "headers": ["X-My-Header: true"]
+                            }
+                        ]
+                    }
+                }]
+            }]
+        }
+
+
+
+.. _k8s_policy:
+
```

Comment:
General comment - we should have a way to be able to import all policies listed in these docs in a test to make sure the guides don't fall out of sync with master.